lighthouse-core/audits/error-logs.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Audits a page to determine if it is calling deprecated APIs.
+ * This is done by collecting console log messages and filtering them by ones
+ * that contain deprecated API warnings sent by Chrome.
+ */
+
+const Audit = require('./audit');
+
+class ErrorLogs extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'error-logs',
+      description: 'Console error logs',
+      helpText: 'In case of errors logged in the console the other metrics might be impacted.',
+      optimalValue: '0',
+      failureDescription: 'Error logs found in the console',
+      requiredArtifacts: ['ChromeConsoleMessages']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const entries = artifacts.ChromeConsoleMessages;
+
+    const consoleErrors = entries.filter(log => log.entry.level === 'error').map(log => {
```

Comment:
is this first map() necessary? seems like all of this work could be done in the tableRows ones.